Replicator/Replicator.cc

```diff
@@ -980,4 +1100,78 @@ namespace litecore { namespace repl {
         request->respond(response);
     }
 
+
+    // Handles a "getCollections" request by looking up a peer checkpoint of each collection.
+    void Replicator::handleGetCollections(Retained<blip::MessageIn> request) {
+        auto root = request->JSONBody().asDict();
+        if (!root) {
+            request->respondWithError({"BLIP"_sl, 400, "Invalid getCollections message: no root"_sl});
+            return;
+        }
+        
+        auto checkpointIDs = root["checkpoint_ids"].asArray();
+        if (!checkpointIDs || checkpointIDs.empty()) {
+            request->respondWithError({"BLIP"_sl, 400, "Invalid getCollections message: no checkpoint_ids"_sl});
+            return;
+        }
+        
+        auto collections = root["collections"].asArray();
+        if (!collections || collections.empty()) {
+            request->respondWithError({"BLIP"_sl, 400, "Invalid getCollections message: no collections"_sl});
+            return;
+        }
+        
+        if (checkpointIDs.count() != collections.count()) {
+            request->respondWithError({"BLIP"_sl, 400, "Invalid getCollections message: mismatched checkpoint_ids and collections"_sl});
+            return;
+        }
+        
+        unordered_set<C4Database::CollectionSpec> specs;
+        for (auto &coll : _collections) {
+            specs.insert(coll->getSpec());
```

Comment:
Perhaps we should add a check that the collection has at least one Passive element (push/pull).

Reply:
Is this required? Do we allow to mix passive and active configuration? If we do not allow that this should be validated when the Replicator is created so that we don't have mixes.

Reply:
From slack, this situation could happen within LiteCore's tests.